app/Community/Listeners/WriteUserActivity.php

```diff
@@ -66,9 +66,14 @@ public function handle(object $event): void
                 $subjectId = $event->achievement->id;
                 $context = $event->hardcore ? 1 : null;
 
-                // Manual unlocks should not update the user's LastLogin.
-                $unlock = $user->playerAchievements()->firstWhere('achievement_id', $subjectId);
-                $updateLastLogin = ($unlock?->unlocker_id ?? 0) === 0;
+                if (!System::isGameSystem($event->achievement->game->system->id)) {
+                    // event unlocks should not update the user's LastLogin
+                    $updateLastLogin = false;
+                } else {
+                    // Manual unlocks should not update the user's LastLogin.
+                    $unlock = $user->playerAchievements()->firstWhere('achievement_id', $subjectId);
+                    $updateLastLogin = ($unlock?->unlocker_id ?? 0) === 0;
```

Comment:
Minor (optional) suggestion while we're in here, this may make the code easier to understand for a future maintainer:
```php
$unlock = $user->playerAchievements()->firstWhere('achievement_id', $subjectId);
$isManualUnlock = $unlock && $unlock->unlocker_id !== 0;
$updateLastLogin = !$isManualUnlock;
```